### PR TITLE
modified swapping in Algorithm.swift  and starting of cell swap in AlgorithmDetailPresenter.swift

### PR DESCRIPTION
**Pull request by AnanthaKrish**

### CODE REVIEW

File: SortingAlgorithms/Scenes/AlgorithmDetail/AlgorithmDetailPresenter.swift [hunk 1]
```diff
@@ -30,24 +30,13 @@ final class AlgorithmDetailPresenter {
...
-            
-            view?.setNavigationBarTitle(algorithm.title)
-            view?.updateDataSource(datasource)
-            view?.swapCell(x0: swaps[0].x0,x1: swaps[0].x1)
-        }
-    }
-    
-    func swapDidComplete() {
-        swaps.removeFirst()
+    func startSwap() {
```
Comment (victorpanitz) [inline SortingAlgorithms/Scenes/AlgorithmDetail/AlgorithmDetailPresenter.swift]: As the presentation pattern is MVP, we should consider the view dummy. Then, it should not tells presenter what presenter is supposed to and but only delegate some states and interactions.

Comment (victorpanitz) [reply SortingAlgorithms/Scenes/AlgorithmDetail/AlgorithmDetailPresenter.swift]: An approach is to keep the methods describing the real behavior and encapsulate the startSwap() as an private method.

Comment (victorpanitz) [reply SortingAlgorithms/Scenes/AlgorithmDetail/AlgorithmDetailPresenter.swift]: ex: 

```Swift
func viewDidAppear(){
    startSwap()
}

func swapDidComplete() {
   startSwap()
}

private func startSwap()...
```

**Review by victorpanitz — COMMENTED** [review]
Hello AnanthaKrish, first, thank you for your contribution. I've reviewed some points which are supposed to be really easy to refactor. The mentioned points are related to the presentation pattern we are using here. After these changes I'd be happy to announce you as a contributor of that project :)

**Comment by AnanthaKrish** [comment]
@victorpanitz Hi, that's right. I think making `startSwap()` as private method and calling it from `viewDidAppear()` and  `swapDidComplete()` would be the best approach. I have updated the code . Please review it and let me know.

**Review by victorpanitz — APPROVED** [review]
Nice! I'd like to say the app is currently waiting to be reviewed at App Store :) You're now contributing to an app that will be reaching people at the store also 👍

**Comment by AnanthaKrish** [comment]
That's nice.  Thank you @victorpanitz

**Pull request merged**